toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -121,7 +122,9 @@ public static void closeScope(Object name) {
    */
 
   public static void reset() {
-    MAP_KEY_TO_SCOPE.clear();
+    for (Object name : new ArrayList<>(MAP_KEY_TO_SCOPE.keySet())) {
+      closeScope(name);
+    }
```

Comment:
we should still clear the map, this is used not to leak during tests. Even though we use a weak reference, we should clear the map. The leak of scopes without references to contexts (on Android) is not much better than leaking a context. 

TP should not reference the scopes, the errors in leaks will come from devs still referencing and using them in some way.